src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -75,99 +73,42 @@ internal static SafeX509CrlHandle DownloadCrl(string uri, ref TimeSpan remaining
 
         private static byte[] DownloadAsset(string uri, ref TimeSpan remainingDownloadTime)
         {
-            if (remainingDownloadTime <= TimeSpan.Zero)
+            if (remainingDownloadTime > TimeSpan.Zero)
             {
-                return null;
-            }
-
-            List<byte[]> dataPieces = new List<byte[]>();
-
-            using (Interop.Http.SafeCurlHandle curlHandle = Interop.Http.EasyCreate())
-            {
-                GCHandle gcHandle = GCHandle.Alloc(dataPieces);
-                Interop.Http.SafeCallbackHandle callbackHandle = new Interop.Http.SafeCallbackHandle();
-
+                object httpClient = null;
                 try
                 {
-                    Interop.Http.EasySetOptionString(curlHandle, Interop.Http.CURLoption.CURLOPT_URL, uri);
-                    Interop.Http.EasySetOptionLong(curlHandle, Interop.Http.CURLoption.CURLOPT_FOLLOWLOCATION, 1L);
-
-                    IntPtr dataHandlePtr = GCHandle.ToIntPtr(gcHandle);
-                    Interop.Http.RegisterReadWriteCallback(
-                        curlHandle,
-                        Interop.Http.ReadWriteFunction.Write,
-                        s_writeCallback,
-                        dataHandlePtr,
-                        ref callbackHandle);
-
-                    Stopwatch stopwatch = Stopwatch.StartNew();
-                    Interop.Http.CURLcode res = Interop.Http.EasyPerform(curlHandle);
-                    stopwatch.Stop();
-
-                    // TimeSpan.Zero isn't a worrisome value on the subtraction, it only
-                    // means "no limit" on the original input.
-                    remainingDownloadTime -= stopwatch.Elapsed;
-
-                    if (res != Interop.Http.CURLcode.CURLE_OK)
+                    Type httpClientType = Type.GetType("System.Net.Http.HttpClient, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                    if (httpClientType != null)
                     {
-                        return null;
+                        // public Task<byte[]> GetByteArrayAsync(string requestUri)
+                        MethodInfo getByteArrayAsync = httpClientType.GetMethod("GetByteArrayAsync", new Type[] { typeof(string) });
+                        if (getByteArrayAsync != null)
+                        {
+                            httpClient = Activator.CreateInstance(httpClientType);
+
+                            Stopwatch stopwatch = Stopwatch.StartNew();
+                            byte[] result = ((Task<byte[]>)getByteArrayAsync.Invoke(httpClient, new object[] { uri })).GetAwaiter().GetResult();
+                            stopwatch.Stop();
+
+                            // TimeSpan.Zero isn't a worrisome value on the subtraction, it only
+                            // means "no limit" on the original input.
+                            remainingDownloadTime -= stopwatch.Elapsed;
```

Comment:
Oops, I just noticed the original code was factoring in this time even in the case of failure; will fix.